sway-lsp/tests/lib.rs

```diff
@@ -913,95 +913,6 @@ async fn go_to_definition_for_traits() {
     shutdown_and_exit(&mut service).await;
 }
 
-#[tokio::test]
```

Comment:
I've opened #4482 as a note to add this test back in, but I'll do that later so we don't hold this PR up.

Reply:
I thought this test was testing for StorageReassignment since that's the only place the `storage` symbol kind was used and I removed that now that's it not used. 🤔

Reply:
I just pulled down and installed this branch and it does seem like there has been a regression somewhere. Previously this test was testing go to definition for `storage` , `var1`, and `x`. 

```rust
storage.var1.x = 42;
```

I installed your branch and manually tested go_to for the updated line but it doesn't find anything for each of the 3 tokens. 

```rust
storage.var1.x.write(42);
```

Reply:
Hmm mainly because storage reassignments are completely removed so the LSP logic that I removed now has to be rewritten with the new API?

Reply:
All good, I'm happy to add this functionality back in a follow up PR, no need to hold this up.